distributed/tests/test_nanny.py

```diff
@@ -330,6 +330,25 @@ async def test_environment_variable_config(c, s, monkeypatch):
         assert results[n.worker_address]["D"] == "123"
 
 
+@gen_cluster(
+    nthreads=[("", 1)],
+    client=True,
+    Worker=Nanny,
+    config={
+        "distributed.nanny.pre-spawn-environ": {"PRE-SPAWN": 1},
+        "distributed.nanny.environ": {"POST-SPAWN": 2},
+    },
+)
+async def test_environment_variable_pre_post_spawn(c, s, n):
+    assert n.env == {"PRE-SPAWN": "1", "POST-SPAWN": "2"}
+    results = await c.run(lambda: os.environ)
+    assert results[n.worker_address]["PRE-SPAWN"] == "1"
+    assert results[n.worker_address]["POST-SPAWN"] == "2"
+
+    del os.environ["PRE-SPAWN"]
```

Comment:
in a `finally`? not a big deal if it leaks to other tests though

Reply:
a bit overkill IMHO